wagtail/tests/test_page_model.py

```diff
@@ -442,6 +442,20 @@ def test_request_routing(self):
         (found_page, args, kwargs) = homepage.route(request, ["events", "christmas"])
         self.assertEqual(found_page, christmas_page)
 
+    def test_cached_parent_obj_set(self):
```

Comment:
Since we're only optimising internals here, I was thinking we could probably get away with adding a `self.assertNumQueries()` to an existing test. What do you think @Nigel2392?